### DISCUSSION TITLE
How to build a knowledge base webapp similar to https://nextjs.org/docs

### DISCUSSION THREAD

**Discussion by truongsinh** [vercel/next.js#39347] [Help] [answered]
I searched for templates/examples, and see mostly blog, not knowledge base (specifically, I like the search bar with real time fuzzy search result). I clone the whole https://github.com/vercel/next.js and know that docs are generated from MDX file in `docs` and `packages/next/pages/_document.tsx`. I did try `pnpm run build --filter next`, but it runs the build, not the dev (with fast refresh) of the docs, while `pnpm run dev --filter next` is not a valid command.

Any guidance how to achieve a knowledge base similar  to https://nextjs.org/docs with that search bar is highly appriciated.

**Comment by falsepopsky** [comment]
it is [React Autosuggest](https://react-autosuggest.js.org/)

**Comment by truongsinh** [comment, marked as answer]
I actually found https://github.com/shuding/nextra , and the version 1 use algolia, but version 2 use flexsearch ♥
